File: day2.erl
-module(day2).

-export([part1/1, part2/1]).


get_depth_height([], Depth, Distance) ->
    Depth * Distance;
get_depth_height([[Direction, Value]|Rest], Depth, Distance) ->
    case Direction of
        forward -> 
            get_depth_height(Rest, Depth, Distance + Value);
        down ->
            get_depth_height(Rest, Depth + Value, Distance);
        up ->
            get_depth_height(Rest, Depth - Value, Distance)
    end.

get_aim([], Depth, Distance, _) ->
    Depth * Distance;
get_aim([[Direction, Value]|Rest], Depth, Distance, Aim) ->
    case Direction of
        forward -> 
            get_aim(Rest, Depth + (Aim * Value), Distance + Value, Aim);
        down ->
            get_aim(Rest, Depth, Distance, Aim  + Value);
        up ->
            get_aim(Rest, Depth, Distance, Aim  - Value)
    end.


part1(Lines) ->
    get_depth_height(util:parse(Lines, [atom, integer]), 0, 0).

part2(Lines) ->
    get_aim(util:parse(Lines, [atom, integer]), 0, 0, 0).

File: Eva's Garbage Fire/day2.py
print('hewwo :3')


def part1():
    file = open('day2.txt', 'r')

    rise = 0
    sink = 0
    forwards = 0
    result = 0

    for line in file:
        line.split(' ')
        print(line.split(' '))

        direction = line.split(" ")[0]
        value = line.split(" ")[1]

        if direction == 'forward':
            forwards = forwards + int(value)

        if direction == 'up':
            rise = rise + int(value)

        if direction == 'down':
            sink = sink + int(value)

        depth = sink - rise
        result = depth * forwards

    print(result)
    file.close()


part1()


def part2():
    file = open('day2.txt', 'r')

    aim = 0
    forwards = 0
    result = 0
    depth = 0

    for line in file:
        print(line.split(' '))

        direction = line.split(" ")[0]
        X = int(line.split(" ")[1])

        if direction == 'up':
            aim = aim - X

        elif direction == 'down':
            aim = aim + X

        elif direction == 'forward':
            forwards = forwards + X
            depth = depth + aim*X

        result = depth * forwards

    print(result)
    file.close()


part2()




